### ISSUE TITLE
Amadron custom trades and Arial Interface issue 

### ISSUE THREAD

**Issue by BiletoothMC** [TeamPneumatic/pnc-repressurized#578]
Hi, I am trying to setup a system using both the Amadron tablet and the Arial Interface combined with other mods from a modpack. I've got it working under certain conditions but having some questions on certain use cases. I hope this is the right place to post this if not I will delete it. 

What I want to do:
Setup a system in which I can wirelessly request a certain item present in my storage from anyplace in the world and have it appear in my inventory through the Arial Interface. 

What I currently have working:
I have a Yabba barrel storage setup linked together with a Yabba barrel connector. The connector is designated as a storage frame in my logistics network. With the Amadron tablet I set up a custom trade requesting an item from my storage that I want to receive wirelessly. I make a trade with myself using diamonds as a currency present in the storage, with a GPS I designated the requested drop of item in a chest that outputs in the Arial interface. I've set the barrels up so they keep exactly 1 diamond stocked in the Yabba Barrel. While the custom trade is up, Amadron drones will continuously come to pick up the 1 diamond and add it to the Amadron trade stock which then triggers the Yabba barrel to add another diamond which triggers a new Amadrone to come pick it up again. To combat this (stop that process) I delete the custom trade as soon as I received the requested item. 
Now this seems to work for which I am happy! My problem occurs when I get some distance away from the setup or go to another dimensions (so presumably unloading the chunk) which causes the system to not work anymore. 
My question is if there is any way I can make this work while not near the system. Would loading the chunk fix it or does it have to be specifically player loader. Or is there another way to achieve what I want that I'm not seeing?
Thanks!

**Comment by BiletoothMC** [comment]
In addition, would traditional vanilla chunkloading with hoppers or one of the several other ways work? Or does it need to be an entity processing chunk? (for which it would need 5x5 loaded chunk with the centre one being entity processing.) And if that's the case, would one entity processing chunk be enough for my system to work?

**Comment by desht** [comment]
First off, congratulations on a clever use of Amadron and the Aerial Interface :)

Anyway, chunkloading should work fine.  I don't know how well the hopper trick works these days, but don't forget that there's a vanilla `/forceload` command now, which will keep chunks loaded.  You might well need to keep multiple chunks loaded since the Amadrones appear from around 30 blocks to the east of the delivery location and fly to it (and I couldn't say offhand if the drone spawning will load the chunk).  I might need to test something similar to this myself to determine exactly how it behaves...

**Comment by BiletoothMC** [comment]
Cool, well I can say it behaves a bit wonkey and it's not the fastest system but hey thats not what I was going for anyway! There's also the possibility when playing on a server someone ninja's my profitable trade but I think thats actually pretty funny. I am currently playing on the testserver of the TechnodeFirmacraft pack so I don't have acces to the /forceload command. There is FTB chunks in the pack but that didn't seem to work in a singleplayer creative world. Either there is something flawed with my FTB chunk settings or I'm out of luck. I will test the forceload command in my SP creative when I get home. Do I close this issue myself or do I post an update of my findings later? I'm very new to GitHub. Thanks!

**Comment by desht** [comment]
Nah, leave it open for now since it raises some interesting questions I actually want to research myself...

**Comment by BiletoothMC** [comment]
Ok so I tried several approaches. I should've mentioned earlier I am playing on a 1.12version of this mod and minecraft version. I am loading the chunks with hopper clocks in a 7x5 radius, which should allow the 3 center clock to be entity loaded. (1 chunk where the system is + 2 in east direction to cover the drones entrance and exit path.) And afterwards I tried loading the same grid of chunks with FTB chunks. Both those test had the same result which was a little different than non loaded. The drones that come to pick up the stocked diamond seem to keep triggering when chunk loaded. However they do not leave the zone but instead stay in the loaded chunks. When I re enter the loaded chunks theyre all their and then suddenly all resume their programa and go stock the picked up diamonds. Placing an order from a distance does not order the item, nor complete it when reentering the loaded chunks.
![ChunkLoadTestHopper](https://user-images.githubusercontent.com/69633361/90668207-00fc0980-e250-11ea-9323-9891f8983ded.png)

**Comment by BiletoothMC** [comment]
I'm an idiot and you can kill me, optifine was present in my pack for shaders. Optifine messes with a lot of stuff including chunk loading. removed optifine from the modlist, tried again and evrything worked, only the deposit chest had to be chunk loaded (1 chunk) and everything worked, even in other dimensions, you can close this if you want, sorry.

**Comment by BiletoothMC** [comment]
Oh wow, I kept trying with Optifine on but turning all chunk settings to default (under optifine options under performance all those that talk about chunks) Read online that often does not fix the issue. But it did it here :) So I can have both the shaders on and order an item from anywhere to anywhere (as long as I have it in my storage and carry around a charged amadron tablet and the gps :P ) cool!

**Comment by desht** [comment]
Excellent.  Sounds like your questions are all answered then?

**Comment by BiletoothMC** [comment]
Mostly yes! The only other 'problem' I had, which I made less of a problem with redstone, is the constant stocking of my personal Amadron trade. It would be cool if you could say to the Amadron system like: 'only ever stock 50' even when theres 100 in the available inventory. Currently if theres 100 items in the designated trading chest, its gonne take it all and add it to that invisible amadron inventory stock. *edit* Invisible inventory is not the right word, you can actually see the quantity in Amadron you can however not see it in a chest or get it out is what I am trying to say*

**Comment by desht** [comment]
Yeah.. I see what you're getting at.  This is something I could possibly add to the 1.16 version of the mod, but I'm not making further changes to 1.12.2 now, I'm afraid.

**Comment by MuteTiefling** [comment]
Actually, rather than changing the way amadron works, I've been thinking it would be pretty cool to have access to an amadron style interface for the programmable remote.

If you want to have a lot of potential requests, either for personal deliveries like this or to build yourself a mini ae2 style crafting system, then the current drop down list system quickly becomes hard to work with since there's no search and no images.

If the remote had an amadron style interface where you can have a custom searchable list with images of the request item, then you could set up this system a bit more easily and have a nicer way of interacting with it.

**Comment by desht** [comment]
I've actually been mulling over a "remote logistics orderer" device which would do something like this.  It would need an Aerial Interface to be useful (and chunkloaded for remote access), but the idea is you'd bind it to an inventory with a Requester Frame on it, and when you use it, it queries all nearby inventories with Storage or Provider frames and presents an AE-style list of available items.  Selecting an item temporarily adds it to the Requester Frame's request list, so nearby drones or logistics modules can deliver it.  If the Requester Frame is on an Aerial Interface, that can go straight to your inventory.  Not sure yet about returning items to the system, but there are possibilities...

**Comment by MuteTiefling** [comment]
I was hoping to go a bit beyond simple requests, actually. I love the idea of automatically building the list though.

I was thinking if the request simply toggled a global variable similar to the existing list function in the remote, a player could potentially program whatever they want from there. That could be a drone selecting items from existing stock, or crafting whatever is needed.

I figure it could set the item in one global and a quantity in another global. Then the program could loop on that quantity to gather up or craft the full command.

Auto building a list of craftables might need some extra prog widgets to communicate with that system though. Maybe a special start widget that marks an api as an on demand craft request. It could then be in the same inventories as other items that all go into the auto populated list.

Not really sure how it might all work, honestly. I've been rolling ideas around in my head for a while now and have been hesitant to post. There's a lot of ways it might be done, but I'd really like it if it wasn't done for you, you know? I love tinkering with prog widgets and an on demand craft request system is probably only missing a couple tiny tweaks on your end to be build able by a player. It's just a matter of figuring out the missing pieces.

Still, just starting out with a way to view and request from an array of chests would be a huge addition to logistics networks. And an enterprising player could always build their own return system by making use of their ender chest and another aerial interface with an active provider frame.
